File: PersianCalendar/src/main/java/com/byagowi/persiancalendar/view/preferences/AthanVolumePreference.java
package com.byagowi.persiancalendar.view.preferences;

import android.content.Context;
import android.util.AttributeSet;

import androidx.preference.DialogPreference;

public class AthanVolumePreference extends DialogPreference {

    public AthanVolumePreference(Context context, AttributeSet attrs) {
        super(context, attrs);
        setDialogIcon(null);
    }

    public int getVolume() {
        return getPersistedInt(1);
    }

    public void setVolume(int volume) {
        final boolean wasBlocking = shouldDisableDependents();
        persistInt(volume);
        final boolean isBlocking = shouldDisableDependents();
        if (isBlocking != wasBlocking) notifyDependencyChange(isBlocking);
    }
}

File: PersianCalendar/src/main/java/com/byagowi/persiancalendar/view/preferences/AthanVolumePreference.kt
package com.byagowi.persiancalendar.view.preferences

import android.content.Context
import android.util.AttributeSet
import androidx.preference.DialogPreference
import com.byagowi.persiancalendar.R

class AthanVolumePreference(context: Context, attrs: AttributeSet) : DialogPreference(context, attrs) {

  var volume: Int
    get() = getPersistedInt(1)
    set(volume) {
      val wasBlocking = shouldDisableDependents()
      persistInt(volume)
      val isBlocking = shouldDisableDependents()
      if (isBlocking != wasBlocking) notifyDependencyChange(isBlocking)
    }

  init {
    dialogLayoutResource = R.layout.preference_volume
    dialogIcon = null
  }
}
